lib/active_model/serializer/adapter/fragment_cache.rb

```diff
@@ -51,20 +51,23 @@ def cached_attributes(klass, serializers)
           non_cached_attributes = attributes - cached_attributes
 
           cached_attributes.each do |attribute|
-            options = serializer.class._attributes_keys[attribute]
-            options ||= {}
-            # Add cached attributes to cached Serializer
-            serializers[:cached].constantize.attribute(attribute, options)
+            add_attribute_to_serializer(attribute, serializers, :cached)
           end
 
           non_cached_attributes.each do |attribute|
-            options = serializer.class._attributes_keys[attribute]
-            options ||= {}
-            # Add non-cached attributes to non-cached Serializer
-            serializers[:non_cached].constantize.attribute(attribute, options)
+            add_attribute_to_serializer(attribute, serializers, :non_cached)
           end
         end
 
+        # Given an attribute and the key of its serializer
+        # 1. Add attribute to the appropriate cached or non-cached Serializer
+        def add_attribute_to_serializer(attribute, serializers, key)
```

Comment:
Thanks for this.  Looking at it, I see other ways it could refactored.. your thoughts? Otherwise, I think I'm happy to merge this. Wanna take a crack at the TODOs as well?

```ruby
def cached_attributes(....)
  ....
  populate_serializer_cache_attributes(:cached, cached_attributes)
  populate_serializer_cache_attributes(:non_cached, non_cached_attributes)
end
  
def serializer_cache_class(serializer_cache_class_type)
  @serializer_cache_classes ||= {}
  @serializer_cache_classes.fetch(serializer_cache_class_type) {
    @serializer_cache_classes[serializer_cache_class_type] =
      serializers[serializer_cache_class_type].constantize
  }
end

def populate_serializer_cache_attributes(serializer_cache_class_type, attributes)
  serializer_klass = serializer.class._attribute_keys
  cache_klass = serializer_cache_class(serializer_cache_class_type)
  attributes.each do |attribute|
    options = serializer_klass._attributes_keys[attribute] || {}
    cache_klass.attribute(attribute, options)
  end
end
```